xcm/xcm-executor/src/traits/on_response.rs

```diff
@@ -94,3 +98,57 @@ impl VersionChangeNotifier for () {
 		false
 	}
 }
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum FinishedQuery<BlockNumber> {
+	Response { response: Response, at: BlockNumber },
+	VersionNotification { origin: MultiLocation, is_active: bool },
+}
+
+/// The possible state of an XCM query response.
+#[derive(Debug, PartialEq, Eq)]
+pub enum QueryResponseStatus<BlockNumber> {
```

Comment:
can be `QueryStatus`, as the pallet's type it represents. We can have some alias for it inside the pallet. And Finished is just Ready. 
but seems like `QueryStatus` was intentionally not put into XCM protocol, only Response and QueryId. A client of `report_outcome` and `take_response`, has a timeout, it cares only if there is response, or there is nothing for now. The rest is not expected. 
Same for the Pay:check_payment, failed, successful and pending for the timeout.